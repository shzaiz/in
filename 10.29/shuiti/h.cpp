#include <bits/stdc++.h>
using namespace std;
int dabiao[1000] = { 4, 7, 44, 77, 47, 74, 444, 777, 477, 747, 774, 744, 474,
    447 };
int x[1002];
int main()
{
    int j = 14;
    for (int n = 1; n <= 1000; n++) {
        for (int i = 0; i < j; i++) {
            if (n == dabiao[i] || n % dabiao[i] == 0) {
                dabiao[j++] = n;
                x[n] = 1;
                break;
            }
        }
    }
    int m;
    cin >> m;
    if (x[m] == 1)
        printf("YES\n");
    else
        printf("NO\n");
    return 0;
}