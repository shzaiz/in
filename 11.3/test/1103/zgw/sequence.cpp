#include <bits/stdc++.h>
using namespace std;
#define N 5000005
#define int unsigned long long
int a[N],n;
signed main(){
	freopen("sequence.in","r",stdin);
	freopen("sequence.out","w",stdout);
	cin>>n;
	int mx = -1, mi = 1e11;
	for(int i = 1;i<=n;i++){
		cin>>a[i];
		mx = max(a[i],mx);
		mi = min(mi, a[i]);
	}
	int cnt = 0;
	for(int i = 1;i<=n;i++){
		if(mi<=a[i] && a[i]<=mx){
			cnt++;
		}
	}
	cout<<mx-mi-n+1<<endl;
	return 0;
}
