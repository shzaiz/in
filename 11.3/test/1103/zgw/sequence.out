18446743973709551616
